tests/core/gui/test_main_window.py

```diff
@@ -185,70 +187,109 @@ def _folder_ready():
 
     await aqtbot.wait_until(_folder_ready)
 
     assert logged_gui.tab_center.count() == 1
 
-    def _mounted():
-        assert core.mountpoint_manager.is_workspace_mounted(f_w.workspace_fs.workspace_id)
 
-    await aqtbot.wait_until(_mounted)
-    await core.mountpoint_manager.unmount_workspace(f_w.workspace_fs.workspace_id)
+@pytest.mark.gui
+@pytest.mark.trio
+@pytest.mark.parametrize("timestamp", [None, "timestamp"])
+async def test_link_file_unmounted(aqtbot, logged_gui_with_files, timestamp, autoclose_dialog):
+    logged_gui, w_w, f_w = logged_gui_with_files
+    await f_w.workspace_fs.sync()
 
-    def _unmounted():
-        assert not core.mountpoint_manager.is_workspace_mounted(f_w.workspace_fs.workspace_id)
+    with freeze_time(DateTime.now().add(seconds=1)):
+        core = logged_gui.test_get_core()
+        if timestamp == "timestamp":
+            timestamp = DateTime.now()
+        url = f_w.workspace_fs.generate_file_link(f_w.current_directory, timestamp)
 
-    await aqtbot.wait_until(_unmounted)
+        logged_gui.add_instance(url.to_url())
 
-    logged_gui.add_instance(url.to_url())
+        def _folder_ready():
+            assert f_w.isVisible()
+            if timestamp is None:
+                assert f_w.table_files.rowCount() == 2
+                folder = f_w.table_files.item(1, 1)
+                assert folder
+                assert folder.text() == "dir1"
+            else:
+                assert f_w.label_role.text() == _(WorkspaceRole.READER)
+                assert isinstance(f_w.workspace_fs, WorkspaceFSTimestamped)
+
+        await aqtbot.wait_until(_folder_ready)
+
+        assert logged_gui.tab_center.count() == 1
+
+        def _mounted(ts):
+            assert autoclose_dialog.dialogs == []
+            assert core.mountpoint_manager.is_workspace_mounted(f_w.workspace_fs.workspace_id, ts)
+
+        await aqtbot.wait_until(lambda: _mounted(None))
+        await core.mountpoint_manager.unmount_workspace(f_w.workspace_fs.workspace_id, None)
+
+        def _unmounted(ts):
+            assert not core.mountpoint_manager.is_workspace_mounted(
+                f_w.workspace_fs.workspace_id, ts
+            )
+
+        await aqtbot.wait_until(lambda: _unmounted(None))
 
-    await aqtbot.wait_until(_mounted)
+        logged_gui.add_instance(url.to_url())
+
+        await aqtbot.wait_until(lambda: _mounted(timestamp))
 
 
 @pytest.mark.gui
 @pytest.mark.trio
-async def test_link_file_invalid_path(aqtbot, autoclose_dialog, logged_gui_with_files):
+@pytest.mark.parametrize("timestamp", [None, "timestamp"])
+async def test_link_file_invalid_path(aqtbot, autoclose_dialog, logged_gui_with_files, timestamp):
     logged_gui, w_w, f_w = logged_gui_with_files
-    url = f_w.workspace_fs.generate_file_link("/unknown")
+    await f_w.workspace_fs.sync()
 
-    logged_gui.add_instance(url.to_url())
+    with freeze_time(DateTime.now().add(seconds=1)):
+        if timestamp == "timestamp":
```

Comment:
Probably sexier to have the parameter as `with_timestamp [True, False]` rather than having a variable named `timestamp` that holds the value `"timestamp"`